File: backend/models/SavedSearch.js
import mongoose from 'mongoose';

const savedSearchSchema = new mongoose.Schema(
  {
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    name: {
      type: String,
      default: 'My Search',
    },
    filters: {
      minPrice: Number,
      maxPrice: Number,
      gender: String,
      roomType: String,
      propertyType: String,
      area: String,
      maxDistance: Number,
      amenities: [String],
    },
    notifyOnNew: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

savedSearchSchema.index({ user: 1 });

const SavedSearch = mongoose.model('SavedSearch', savedSearchSchema);
export default SavedSearch;

File: src/model/SavedSearch.java
package model;

/**
 * Model representing a saved search with filters in the Java service.
 */
public class SavedSearch {
    public String id;
    public String name;
    public User user;
    public Filters filters;

    public SavedSearch() {}

    public static class Filters {
        public Double minPrice;
        public Double maxPrice;
        public String gender;
        public String roomType;
        public String propertyType;
        public Double maxDistance;
        public String area;
        
        public Filters() {}
    }
}
